src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8131,6 +8130,151 @@ public interface ITest
+    T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class Class : [|IInterface<string?>|]
+{
+}",
+@"#nullable enable
+
+interface IInterface<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class Class : IInterface<string?>
+{
+    public string? Method(string? value)
+    {
+        throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnylasisAttributeOnNonNullableValueType()
+        {
+            await TestInRegularAndScript1Async(
+@"interface IInterface<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class<T> : [|IInterface<T>|] where T : struct
+{
+}",
+@"interface IInterface<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class<T> : IInterface<T> where T : struct
+{
+    public T MethodAllowNull(T value)
+    {
+        throw new System.NotImplementedException();
+    }
+
+    public T MethodDisallowNull(T value)
+    {
+        throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnylasisAttributeOnNullableValueType()
+        {
+            await TestInRegularAndScript1Async(
+@"interface IInterface<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class<T> : [|IInterface<T?>|] where T : struct
+{
+}",
+@"interface IInterface<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class<T> : IInterface<T?> where T : struct
+{
+    public T? MethodAllowNull(T? value)
+    {
+        throw new System.NotImplementedException();
+    }
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
```

Comment:
Can we remove annotation here and just have `public T MethodDisallowNull(T value)`?

Reply:
`int` and `Nullable<int>` are incompatible in signature.